pandas/tests/io/pytables/test_pytables.py

```diff
@@ -562,6 +562,25 @@ def check_default_mode():
         check("w")
         check_default_mode()
 
+    def test_store_closed_on_error(self, mocker):
+        """check hdf store is closed whichever error occurs"""
+        mocker.patch("pandas.io.pytables.HDFStore.select").side_effect = AssertionError(
+            "Gaps in blk ref_locs"
+        )
+        df = tm.makeDataFrame()
+        with ensure_clean_path(self.path) as path:
+            df.to_hdf(path, "df")
+            try:
+                pd.read_hdf(path, "df")
+            except AssertionError:  # this is expected, because of our mock
+                pass
+            try:
+                HDFStore(path, mode="w")
+            except ValueError as exc:
+                pytest.fail(
```

Comment:
you don't need to catch something that you are expecting to pass

Reply:
Thanks @jreback for the feedback. I'm not exactly sure about what you meant but if you suggest to remove the `except ValueError`, that was done on purpose in order to fail (`F`) rather than to output an error (`E`).